applications/IgaApplication/custom_conditions/sbm_laplacian_condition.cpp

```diff
@@ -0,0 +1,563 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Nicolò Antonelli
+//                   Andrea Gorgi
+//
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "custom_conditions/sbm_laplacian_condition.h"
+#include "containers/model.h"
+
+namespace Kratos
+{
+    void SBMLaplacianCondition::CalculateAll(
+        MatrixType& rLeftHandSideMatrix,
+        VectorType& rRightHandSideVector,
+        const ProcessInfo& rCurrentProcessInfo,
+        const bool CalculateStiffnessMatrixFlag,
+        const bool CalculateResidualVectorFlag
+    )
+    {
+        std::string boundaryConditionType = this->GetValue(BOUNDARY_CONDITION_TYPE);

```

Comment:
note, we can use `enums` since thy are faster

Reply:
What exactly do you mean by using enums instead of creating a new variable "BOUNDARY_CONDITION_TYPE"?

Reply:
@AlejandroCornejo is suggesting to declare a c++ `enum` (see [here](https://en.cppreference.com/w/cpp/language/enum)) rather than an `std::string` to make the comparisons faster (string comparisons are quite slow).